### DISCUSSION TITLE
How to extend an endpoint exposed by an extension?

### DISCUSSION THREAD

**Discussion by lucian-torje-siemens** [eclipse-edc/Connector#1222] [Q&A]
Hello,
I need to extend the controller for the assets so that we can add our own logic during mapping - the end goal is to customize the transfer process.

Is there a way to load the extension but still be able to modify its behavior?
Is it possible to replace a transformer for an already loaded resource or the resource itself?

![image](https://user-images.githubusercontent.com/102717569/165460456-953e2fc7-ddb1-46c6-9f74-257dfacd7aa9.png)

Thank you!

**Comment by ndr-brt** [comment]
Hi, currently there's no possibility to replace neither transformer nor the resource, the only way is to write a completely custom module and exclude the original module from the BOM.
What do you mean with "customize the transfer process"? What's the need?

**Reply by lucian-torje-siemens** [reply]
Hi, 
The problem is that we are using presigned URLs which expire after a given time, therefore  we need to create them on the fly. 

The /transferprocess endpoint looks like a good candidate because it gets called when the consumer requests the artifact, so instead of creating a presigned url for the artifact in the initial step, we delay its creation till it is needed/requested.

**Reply by jimmarino** [reply]
Hi, can you explain further the use case you are trying to address? It sounds as if a custom provisioner may be more appropriate for this as the API and transformer modules should probably not have this type of knowledge.

**Reply by lucian-torje-siemens** [reply]
Hi Jim,
A custom provisioner sounds like a good idea - could you please share an example on how it can be done and used?

I've found the following org.eclipse.dataspaceconnector.transfer.provision.http.HttpProvisionerExtension

**Reply by jimmarino** [reply]
@lucian-torje-siemens Sorry for the delay, check out HttpProvisionerExtension.

**Reply by lucian-torje-siemens** [reply]
Thank you @jimmarino. I did checked that example and managed to make some progress - my FilesystemProvisioner gets called and uses dataplane for the transfer.
